src/racing/actors.c

```diff
@@ -1261,105 +728,21 @@ UNUSED s16 D_802B8810[] = {
-    render_actor_shell(camera, matrix, shell);
-}
-
-// A little strange this render banana
-void render_actor_banana(Camera *camera, UNUSED Mat4 arg1, struct BananaActor *banana) {
-    UNUSED s32 pad[2];
-    s32 maxObjectsReached;
-    Vec3s sp7C;
-    Mat4 sp3C;
-
-    f32 temp = is_within_render_distance(camera->pos, banana->pos, camera->rot[1], 0, gCameraZoom[camera - camera1], 490000.0f);
-    if (temp < 0.0f) {
-        func_80297230(camera, (struct Actor *) banana);
-        return;
-    }
-
-    if ((banana->pos[1] > D_8015F6EC + 800.0f)) {
-        func_80297230(camera, (struct Actor *) banana);
-        return;
-    }
-    if (banana->pos[1] < (D_8015F6EE - 800.0f)) {
-        func_80297230(camera, (struct Actor *) banana);
-        return;
-    }
-
-    func_802972B8(camera, (struct Actor *) banana);
-
-    if (banana->state == 5) {
-        mtxf_pos_rotation_xyz(sp3C, banana->pos, banana->rot);
-    } else {
-        sp7C[0] = 0;
-        sp7C[1] = 0;
-        sp7C[2] = 0;
-        mtxf_pos_rotation_xyz(sp3C, banana->pos, sp7C);
-    }
+#include "actors/green_shell/render.inc.c"
 
-    maxObjectsReached = render_set_position(sp3C, 0) == 0;
-    if (maxObjectsReached) { return; }
+#include "actors/blue_and_red_shells/render.inc.c"
 
-    if (banana->state != 5) {
-        gSPDisplayList(gDisplayListHead++, &common_model_banana);
-    } else {
-        gSPDisplayList(gDisplayListHead++, &common_model_flat_banana);
-    }
-}
+#include "actors/banana/render.inc.c"
 
-void update_actor_wario_stadium_sign(struct Actor *arg0) {
-    arg0->rot[1] += 0xB6;
-}
+#include "actors/wario_sign/update.inc.c"
 
-/**
- * If train close activate bell sound according to timing
- **/
-void update_actor_railroad_crossing(struct RailroadCrossing *crossing) {
-    // If train close?
-    if (isCrossingTriggeredByIndex[crossing->crossingId] != 0) {
-        // Timer++
-        crossing->someTimer++;
-        // Reset timer
-        if (crossing->someTimer > 40) {
-            crossing->someTimer = 1;
-        }
-        // Play Bell sound when timer hits 20 or 1.
-        if ((crossing->someTimer == 1) || (crossing->someTimer == 20)) {
-            func_800C98B8(crossing->pos, crossing->velocity, SOUND_ARG_LOAD(0x19, 0x01, 0x70, 0x16));
-        }
-    }
-}
+#include "actors/railroad_crossing/update.inc.c"
 
-void update_actor_mario_raceway_sign(struct Actor *arg0) {
-    if ((arg0->flags & 0x800) == 0) {
-        if ((arg0->flags & 0x400) != 0) {
-            arg0->pos[1] += 4.0f;
-            if (arg0->pos[1] > 800.0f) {
-                arg0->flags |= 0x800;
-                arg0->rot[1] += 1820;
-            }
-        } else {
-            arg0->rot[1] += 182;
-        }
-    }
-}
+#include "actors/mario_sign/update.inc.c"
 
 UNUSED void func_8029ABD4(f32 *pos, s16 state) {
     gNumActors = 0;
-    gActorList[func_8029ED38(pos, 0x0014)].state = state;
+    gActorList[spawn_actor_at_pos(pos, 0x0014)].state = state;
```

Comment:
This actually spawns an actor of type 0x14 which appears to be an unknown actor

Reply:
yeah there is also a update function for the actor 0x14

Reply:
Can we make an enum for this actor but just call it?
`UNKNOWN_ACTOR`

Reply:
I will put all missing id between 1 to 45 so it will be UNKNOWN_ACTOR_0x14

Reply:
If those id's don't exist at all then don't bother.

Reply:
I just create it need to finish doc all update/render function